### ISSUE TITLE
DreamFusion: Text-to-3D using 2D Diffusion

### ISSUE THREAD

**Issue by reyllama** [reyllama/paper-reviews#196]
# TL; DR

- Efficiently optimize a text-dependent NeRF using a text-to-image 2D diffusion model to perform text-to-3D.

# Motivation

- Directly generating 3D assets from text prompts requires a large volume of (text, 3D) dataset, which is not available.

# Formulation

![image](https://user-images.githubusercontent.com/46308407/205050976-d9a74cd4-bf8e-43bc-9226-aee0bc4460d0.png)

### Training Objective

![image](https://user-images.githubusercontent.com/46308407/205051167-2042eb34-d972-49cb-883b-fc38c67db916.png)

- This naive diffusion loss computes U-Net Jacobian, which directly uses the noise prediction given (z_t, t) and incurs computational/numerical burden.
- The authors choose to omit this term and obtain the following *score distillation objective*, which boils down to the gradient of a weighted probability density distillation loss.

![image](https://user-images.githubusercontent.com/46308407/205051431-05034a6d-39b5-48e9-8557-d1943fd07fcd.png)
![image](https://user-images.githubusercontent.com/46308407/205051456-f6ce7a6d-6634-45ce-bfaa-0b31d7e6f564.png)

- In the implementation level, this implies that we no longer use the direct noise prediction (at step t) for loss computation.
- Instead, we obtain a tensor (noise estimate) in anyway we want (as we no longer need its computational graph) and compute the update direction (\hat{\eps}-\eps), which saves computation and improves stability.

### Training Procedure

- Randomly sample camera pose and light (the authors use shading instead of radiance).
- Volume-render a 2D image.
- Compute diffusion loss. (For this, we add noise to the rendered image, denoise it to clean sample using Imagen, and compute the noise closed form w.r.t. the noised rendered image)
- Back-propagate this loss to NeRF parameters (DIP).
- Note that there are much more technical details in shading, view-aware prompt composition and camera/light sampling to enhance the output quality that I skipped for brevity.

# Evaluation

![image](https://user-images.githubusercontent.com/46308407/205053409-96f4138f-af5b-4800-af86-0d76ba6c2677.png)

- Since there is no corresponding GT to a rendered image, traditional metrics like PSNR and Chamfer Distance are of no use.
- The authors report CLIP R-Precision, which measures the accuracy of the retrieved caption (among distractors) for a rendered image.
- The authors claim that Dream Fields and CLIP-Mesh have a significant upper hand in this benchmark as they use CLIP during the training (*fair argument*).

![image](https://user-images.githubusercontent.com/46308407/205053821-8746b0d1-7e28-4127-b949-630806ec0cef.png)

- Ablation experiment that shows the technical components of DreamFusion helps the model capture fine geometry.

![image](https://user-images.githubusercontent.com/46308407/205053917-6abb1232-40ea-424d-be65-1a3355a1146e.png)

- Interactive composition of a 3D asset.

# Inspiration

- I was thinking of ideas to leverage the strong generative prior diffusion model (e.g., stable diffusion) to generalize NeRF models when I found this paper. Similar motivation, but much better (and quicker) I guess.
- In idea level, this is more similar to 3D-aware generative model families such as pi-GAN, GRAF and GIRAFFE. They incorporate implicit 3D representations (and volume rendering) in their pipeline (as a plug-in component) to inject 3D knowledge to the generator.
- Their key insight (in my opinion) lies in simplification of the diffusion loss formulation, for which they diffuse and reconstruct (denoise) the rendered image to compute the noise prediction analytically. Smart idea.
